storage/remote/queue_manager.go

```diff
@@ -51,66 +42,73 @@ type StorageClient interface {
 	Name() string
 }
 
+type StorageQueueManagerConfig struct {
+	QueueCapacity     int           // Number of samples to buffer per shard before we start dropping them.
+	Shards            int           // Number of shards, i.e. amount of concurrency.
+	MaxSamplesPerSend int           // Maximum number of samples per send.
+	BatchSendDeadline time.Duration // Maximum time sample will wait in buffer.
+}
+
+var defaultConfig = StorageQueueManagerConfig{
```

Comment:
One last nit: if something is called `defaultConfig`, I would expect that the constructor takes a pointer to a config and if it's nil, use the default config automatically, rather than having every caller pass it in.